.github/workflows/backend.yml

```
@@ -0,0 +1,88 @@
+name: KFP Tekton backend unit tests
+
+on:
+  push:
+    branches: [master]
+
+  # Run tests for any PRs which change the backend code
+  pull_request:
+    paths:
+      - 'go.mod'
+      - 'backend/**'
+      - 'scripts/deploy/github/**'
+      - 'manifests/kustomize/**'
+
+env:
+  GITHUB_ACTION: "true"
+  SETUPTOOLS_USE_DISTUTILS: "stdlib"
+
+jobs:
+  run-go-unittests:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: 1.20.x
+    - name: Checkout code
+      uses: actions/checkout@v4
+    - name: "run go unit tests"
+      run: go test -v -cover ./backend/...
+  backend-integration:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Create k8s Kind Cluster
+        uses: container-tools/kind-action@v2
+        with:
+          cluster_name: kfp-tekton
+          kubectl_version: v1.29.2
+          version: v0.17.0
```

I'm not sure kind 1.17 has the bug fixes for the k8s 1.29 cluster deployment. Maybe we either upgrade kind to 0.22 which has k8s 1.29 support or just run `node_image: kindest/node:v1.26.3` to minimize the deployment variable changes.
```suggestion
          version: v0.22.0
```